psl-core/src/main/java/org/linqs/psl/reasoner/term/online/OnlineTermStore.java

```diff
@@ -47,13 +53,23 @@
     protected List<Integer> activeVolatilePages;
     protected Integer nextTermPageIndex;
     protected Integer nextVolatilePageIndex;
+    protected Map<Rule, List<Integer>> rulePageMapping;
+    protected Map<Rule, Boolean> activatedRules;
```

Comment:
There seems to be a good amount of complexity around tracking active pages/rules (and I have concerns about possible issues in the correctness of the handling of some mixed pages).

What if we just limited it to one rule per page at that StreamingTermStore (and the StreamingGroundingIterator) level?
Then we can just ignore or destroy entire pages.

Reply:
One rule per page is ensured by the primeNextRuleIndex() method in the onlineGroundingIterator class, so there are no mixed pages.